validation/src/main/scala/hmda/validation/engine/lar/macro/LarMacroEngine.scala

```diff
@@ -0,0 +1,42 @@
+package hmda.validation.engine.lar.`macro`
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.api.ValidationApi
+import hmda.validation.engine.lar.LarCommonEngine
+import hmda.validation.engine.{ Macro, ValidationErrorType }
+import hmda.validation.rules.AggregateEditCheck
+import hmda.validation.rules.lar.`macro`.MacroEditTypes.LoanApplicationRegisterSource
+import hmda.validation.rules.lar.`macro`.{ Q007, Q008 }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+trait LarMacroEngine extends LarCommonEngine with ValidationApi {
+
+  def checkMacro(larSource: LoanApplicationRegisterSource)(implicit system: ActorSystem, materializer: ActorMaterializer, ec: ExecutionContext): Future[LarSourceValidation] = {
+    Future.sequence(
+      List(
+        Q007,
+        Q008
+      ).map(checkAggregate(_, larSource, "", Macro))
+    )
+      .map(checks => validateAll(checks, larSource))
+
+  }
+
+  private def checkAggregate(
+    editCheck: AggregateEditCheck[LoanApplicationRegisterSource, LoanApplicationRegister],
+    input: LoanApplicationRegisterSource,
+    inputId: String,
```

Comment:
At this point in time, I don't think so. That `id` is there to identify individual loan and respondent ids at the record level. For groups of `LARs` we haven't come up with an id to pass yet. It's like this to reuse the existing functionality of `convertResult`. I'm ok with refactoring this, but I also don't think this should hold this PR